src/Yarp.ReverseProxy.Swagger/ReverseProxyDocumentFilter.cs

```diff
@@ -205,5 +214,48 @@ DocumentFilterContext context
 
             return validRoutes;
         }
+
+        private void ApplySwaggerTransformation(List<OperationType> operationKeys, KeyValuePair<string, OpenApiPathItem> path, IEnumerable<RouteConfig> routes)
```

Comment:
I suggest to use method below instead of your one. I delete excess loop iterations. Also I think exception throwing is not necessary.

```
private void ApplySwaggerTransformation(
    List<OperationType> operationKeys,
    KeyValuePair<string, OpenApiPathItem> path,
    string clusterKey)
{
    var factories = _factories
        ?.Where(x => x is ISwaggerTransformFactory)
        .ToList();
    
    if (factories == null) return;
    
    foreach (var operationKey in operationKeys)
    {
        path.Value.Operations.TryGetValue(operationKey, out var operation);
        
        var transforms = _config.Routes
            .Where(x => x.Value.ClusterId == clusterKey)
            .Where(x => x.Value.Transforms != null)
            .SelectMany(x => x.Value.Transforms)
            .ToList();
        
        foreach (var swaggerFactory in factories.Select(factory => factory as ISwaggerTransformFactory))
        {
            foreach (var transform in transforms)
            {
                swaggerFactory?.Build(operation, transform);
            }
        }
    }
}
```

Reply:
Perfect. Fixed.